WSL/disk-space.md

```diff
@@ -0,0 +1,207 @@
+```
+
+The virtual drive (ext4.vhdx) for this Linux distribution has now successfully been expanded to the new size.  
+
+> [!IMPORTANT]
+> We recommend that you do not modify, move, or access the WSL related files located inside of your `AppData` folder using Windows tools or editors. Doing so could cause your Linux distribution to become corrupted. If you would like to access your Linux files from Windows, that is possible via the path `\\wsl$\<distribution-name>\`. Open your WSL distribution and enter `explorer.exe .` to view that folder. To learn more, see the blog post: [Accessing Linux files from Windows](https://devblogs.microsoft.com/commandline/whats-new-for-wsl-in-windows-10-version-1903/#accessing-linux-files-from-windows).
+
+## How to repair a VHD mounting error
+
+If you encounter an error related to "mounting the distribution disk", this could be due to a sudden shutdown or power outage and may result in the Linux distribution VHD being switched to *read-only* to avoid data loss. You can repair and restore the distribution using the `e2fsck` Linux command by following the steps below.
+
+### Use the lsblk command to identify the block device name
+
+When WSL 2 installs a Linux distribution, it is mounting the distribution as a Virtual Hard Disk (VHD) with it's own file system. Linux refers to these hard drives as "block devices" and you can view information about them by using the `lsblk` command.
+
+To find the names of the block devices currently being used by WSL 2, open your distribution and enter the command: `lsblk`. (Or open PowerShell and enter the command: `wsl.exe lsblk`.) The output will look something like this:
+
+```bash
+NAME MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+sda    8:0    0 363.1M  1 disk
+sdb    8:16   0     8G  0 disk [SWAP]
+sdc    8:32   0   1.5T  0 disk
+sdd    8:48   0     1T  0 disk /mnt/wslg/distro
+```
+
+Information about the block device includes:
+
+- **NAME**: The name assigned to the device will be sd[a-z], referring to the SCSI Disk with a letter designation for each disk being used. `sda` is always the system distribution.
+- **MAJ:MIN**: Represents numbers used by the Linux kernel to internally identify the devices with the first number representing the device type (8 is used for Small Computer System Interface/SCSI disks).
+- **RM**: Let's us know if the device is removable (1) or not (0).
+- **SIZE**: Total size of the volume.
+- **RO**: Let's us know if the device is read-only (1) or not (0).
+- **TYPE**: Refers to the device type (disk in this case).
+- **MOUNTPOINTS**: Refers to the current directory on the files system where the block device is located (SWAP is for preconfigured inactive memory so no mountpoint).
+
+### Read-only fallback error
+
+If WSL encounters a "mounting error" when opening a Linux distribution, the distribution may be set as read-only (RO = 1) as a fallback. If that happens, the distribution may display the following error during startup:
```

Comment:
I think we should remove the `(RO = 1)` because might create confusion between read-only devices (what lsblk shows) and read-only mountpoints (which is what this section is about)

Reply:
Removed. Thanks for pointing out this differentiation!